### DISCUSSION TITLE
TBX on Alpine Linux

### DISCUSSION THREAD

**Discussion by orekerete** [watermint/toolbox#641] [Q&A]
Firstly thanks for the hard work to put this together.  I have been successfully using this on my x86 Debian Linux distro and just noticed you have released a Linux Arm version.  Do you know if it works on Alpine Linux as I been trying to get it to work on my Raspberry Pi running Alpine with no luck.

**Comment by watermint** [comment]
Yes it works on Alpine Linux. But you need to install below packages to run.

```
apk add gcompat
```

I'm not sure if it works on Raspberry Pi (I don't have one). Because this tool targeted for PC environment (4GB+ memory). Memory/storage footprint is designed for that. Memory footprint varies on command, but some command requires 1GB+ heap memory.

**Reply by orekerete** [reply]
Thanks for the prompt response.  The RPi I am testing with has 4GB RAM so hopefully it meets the requirements, and I added the packages and got a bit further but still with errors on a straight run of the executable without any parameters.
./tbx: line 0: �F@�@8@@@@@@����@�@dd@@�N��N�P�PP��͸��������: not found
./tbx: line 0:0�
                Q�t�e@�: not found
./tbx: line 0: @_@P: not found
./tbx: line 0: @�: not found
./tbx: line 1: ELF: not found
./tbx: line 3: syntax error: unexpected "("

Any suggestions would be appreciated.  Thanks.

**Reply by watermint** [reply]
I have no idea why that happened. But it look like your shell may execute program as shell script not binary executable.

**Reply by orekerete** [reply]
I am happy to report that I have it working on my Raspberry Pi.  I had previously copied the Linux executable, and even though I had replaced it with the Linux Arm version I forgot to issue the Apline command lbu commit in order to save the changes, thus the previous error was due to me still running the wrong executable.  Thanks again for your hard work and subsequent assistance.
